web/static/js/components/thesis-editor.js

```diff
@@ -312,6 +317,38 @@ class ThesisEditor extends React.Component {
     this.setState({importProgress: progress})
   }
 
+  checkForDuplicateContentBlocks () {
+    const editors = this.allContentEditors()
+
+    const names = {}
+    for (let i = 0; i < editors.length; i++) {
+      const name = editors[i].dataset.thesisContentId
+      if (names[name]) {
+        names[name] += 1
+      } else {
+        names[name] = 1
+      }
```

Comment:
Keep in mind that you're mutating `names[name]` and then assigning it again to the same value by doing this.

![image](https://user-images.githubusercontent.com/1479215/27895735-abcfec38-61c9-11e7-857b-1240872fd4f5.png)

This is generally considered a practice to avoid. I'd do this instead:

```javascript
names[name] = names[name] ? names[name] + 1 : 1
```